cpp/src/arrow/buffer-builder.h

```diff
@@ -45,23 +45,23 @@ class ARROW_EXPORT BufferBuilder {
 
   /// \brief Resize the buffer to the nearest multiple of 64 bytes
   ///
-  /// \param elements the new capacity of the of the builder. Will be rounded
+  /// \param new_capacity the new capacity of the of the builder. Will be rounded
   /// up to a multiple of 64 bytes for padding
   /// \param shrink_to_fit if new capacity is smaller than the existing size,
   /// reallocate internal buffer. Set to false to avoid reallocations when
   /// shrinking the builder.
   /// \return Status
-  Status Resize(const int64_t elements, bool shrink_to_fit = true) {
+  Status Resize(const int64_t new_capacity, bool shrink_to_fit = true) {
     // Resize(0) is a no-op
-    if (elements == 0) {
+    if (new_capacity == 0) {
       return Status::OK();
     }
     int64_t old_capacity = capacity_;
 
     if (buffer_ == NULLPTR) {
-      ARROW_RETURN_NOT_OK(AllocateResizableBuffer(pool_, elements, &buffer_));
+      ARROW_RETURN_NOT_OK(AllocateResizableBuffer(pool_, new_capacity, &buffer_));
     } else {
-      ARROW_RETURN_NOT_OK(buffer_->Resize(elements, shrink_to_fit));
+      ARROW_RETURN_NOT_OK(buffer_->Resize(new_capacity, shrink_to_fit));
     }
     capacity_ = buffer_->capacity();
     data_ = buffer_->mutable_data();
```

Comment:
It would be good to evaluate whether we really need the `memset` below, since it sounds like it can be wasteful for large upsizes. Ideally, only the padding area needs to be zeroed (see `ResizePoolBuffer` in `buffer.cc`).

Reply:
I'm looking into this right now